test/IntegrationTests/WcfTestsBase.cs

```diff
@@ -72,4 +74,25 @@ public void Dispose()
         Output.WriteLine($"Exit Code: " + _serverProcess.Process.ExitCode);
         Output.WriteResult(_serverProcess);
     }
+
+    private async Task WaitForServer()
+    {
+        using var tcpClient = new TcpClient();
+        const int tcpPort = 9090;
+
+        while (true)
+        {
+            try
+            {
+                await tcpClient.ConnectAsync("127.0.0.1", tcpPort);
+                Output.WriteLine("WCF Server is running.");
+                return;
+            }
+            catch (Exception)
+            {
+                Output.WriteLine("Waiting for WCF Server to open ports.");
+                await Task.Delay(500);
```

Comment:
What if server fully failed? We will be waiting here forever.
We need to have some timeout/attempts count then fail.